include/solvers/navier_stokes_scratch_data.h

```diff
@@ -769,64 +783,132 @@ class NavierStokesScratchData : public PhysicsScratchDataBase
     fluid_velocity_curls_at_particle_location_3d.resize(number_of_particles);
     fluid_pressure_gradients_at_particle_location.resize(number_of_particles);
 
-    // Evaluate the relevant information at the
-    // quadrature points to do the interpolation.
-    const auto &velocity_cell =
-      typename DoFHandler<dim>::cell_iterator(*this->fe_values.get_cell(),
-                                              &dof_handler);
+    // // Evaluate the relevant information at the
+    // // quadrature points to do the interpolation.
+    // const auto &velocity_cell =
+    //   typename DoFHandler<dim>::cell_iterator(*this->fe_values.get_cell(),
+    //                                           &dof_handler);
 
     fe_values_local_particles.reinit(velocity_cell);
 
     // Calculate all fluid properties at the particle location
     fe_values_local_particles[velocities].get_function_values(
-      previous_solution, fluid_velocity_at_particle_location);
+      vector_solution, fluid_velocity_at_particle_location);
 
     fe_values_local_particles[velocities].get_function_laplacians(
-      previous_solution, fluid_velocity_laplacian_at_particle_location);
+      vector_solution, fluid_velocity_laplacian_at_particle_location);
 
     if constexpr (dim == 2)
       {
         fe_values_local_particles[velocities].get_function_curls(
-          previous_solution, fluid_velocity_curls_at_particle_location_2d);
+          vector_solution, fluid_velocity_curls_at_particle_location_2d);
       }
     else if constexpr (dim == 3)
       {
         fe_values_local_particles[velocities].get_function_curls(
-          previous_solution, fluid_velocity_curls_at_particle_location_3d);
+          vector_solution, fluid_velocity_curls_at_particle_location_3d);
       }
 
     fe_values_local_particles[pressure].get_function_gradients(
-      previous_solution, fluid_pressure_gradients_at_particle_location);
+      vector_solution, fluid_pressure_gradients_at_particle_location);
+  }
 
-    // Create a quadrature for the void fraction that is based on the particle
-    // reference location
-    if (this->interpolated_void_fraction)
-      {
-        FEValues<dim> fe_values_particles_void_fraction(
-          this->fe_values_void_fraction->get_fe(), q_local, update_values);
+  /** @brief Interpolates the void fraction at the locations of the particles.
+   *
+   * @param q_particles Quadrature type object that contains the location of the particles
+   * relative to the cell's frame of reference.
+   *
+   * @param void_fraction_cell The active cell associated with the void fraction DoFHandler
+   *
+   * @param void_fraction_solution The void fraction calculated with one of the methods of
+   * the VoidFractionBase class.
+   */
+
+  template <typename VectorType>
+  void
+  calculate_void_fraction_at_particle_location(
+    const Quadrature<dim>                                &q_particles,
```

Comment:
Same as above, good time to have a more verbose adequate name for the quadrature.